clusterloader2/pkg/prometheus/prometheus.go

```diff
@@ -143,6 +143,10 @@ func NewController(clusterLoaderConfig *config.ClusterLoaderConfig) (pc *Control
 		// Backward compatibility.
 		clusterLoaderConfig.PrometheusConfig.ScrapeKubeProxy = mapping["PROMETHEUS_SCRAPE_KUBE_PROXY"].(bool)
 	}
+	if !clusterLoaderConfig.ClusterConfig.Provider.Features().ShouldScrapeKubeProxy {
```

Comment:
Please merge this with the if above:
```
if _, exists := mapping["PROMETHEUS_SCRAPE_KUBE_PROXY"]; !exists {
  clusterLoaderConfig.PrometheusConfig.ScrapeKubeProxy = clusterLoaderConfig.ClusterConfig.Provider.Features().ShouldScrapeKubeProxy
  mapping["PROMETHEUS_SCRAPE_KUBE_PROXY"] = clusterLoaderConfig.PrometheusConfig.ScrapeKubeProxy
} else {
  // Backward compatibility
  clusterLoaderConfig.PrometheusConfig.ScrapeKubeProxy = mapping["PROMETHEUS_SCRAPE_KUBE_PROXY"].(bool)
}
```

If someone is explicitly requesting that - we should honor that no matter what provider it is.